util/alloydbutil/options.go

```diff
@@ -93,10 +95,18 @@ func WithIAMAccountEmail(email string) Option {
 	}
 }
 
+// WithUserAgent returns an Option that sets the User-Agent.
+func WithUserAgent(ua string) Option {
+	return func(p *engineConfig) {
+		p.userAgents = append(p.userAgents, ua)
+	}
+}
+
 func applyClientOptions(opts ...Option) (engineConfig, error) {
 	cfg := &engineConfig{
 		emailRetreiver: getServiceAccountEmail,
 		ipType:         "PUBLIC",
+		userAgents:     []string{defaultUserAgent},
```

Comment:
It doesn't look like this gets ultimately passed into the NewDialer options (https://github.com/averikitsch/langchaingo/blob/9e8447a7a2feb3f43a6553825177f58893920d93/util/cloudsqlutil/engine.go#L48)

@kurtisvg can you review?

Reply:
oh, I see, let me fix it